utils/ixdtf/src/parsers/records.rs

```diff
@@ -0,0 +1,185 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! Parse records for `ixdtf`'s parsers to target.
+
+use alloc::string::String;
+
+/// An `IsoParseRecord` is an intermediary record returned by ISO parsing functions.
+///
+/// `IsoParseRecord` is converted into the ISO AST Nodes.
+#[non_exhaustive]
+#[derive(Default, Debug, PartialEq)]
+pub struct IsoParseRecord {
+    /// Parsed Date Record
+    pub date: DateRecord,
+    /// Parsed Time
+    pub time: Option<TimeRecord>,
+    /// Parsed `TimeZone` data (UTCOffset | IANA name)
+    pub tz: Option<TimeZone>,
+    /// The parsed calendar value.
+    pub calendar: Option<String>,
+}
```

Comment:
Suggestion: as noted elsewhere, I think this should end up borrowing more data, and it should probably have a place to put unknown annotations, like this

```rust
pub struct IsoParseRecord<'a> {
    /// Parsed Date Record
    pub date: DateRecord,
    /// Parsed Time
    pub time: Option<TimeRecord>,
    /// Parsed `TimeZone` data (UTCOffset | IANA name)
    pub tz: Option<TimeZone<'a>>,
    /// The parsed calendar value.
    pub calendar: Option<&'a str>,
    /// Any additional annotations beyond time zone and calendar
    pub annotations: Vec<Annotation<'a>>,
}

pub struct Annotation<'a> {
    pub key: &'a str,
    pub values: &'a str,
    pub critical_flag: bool,
}
```

(it also needs a way to record whether the time zone or calendar had the critical flag)

Reply:
That should be doable.

Also, on the subject of the `IsoParseRecord`, the trend mostly seems the `ixdtf` should just parse the string and provide the record to the user. So should the potential `tz` ambiguity be provided to the user as well.

```rust
pub struct IsoParseRecord<'a> {
    /// Parsed Date Record
    pub date: DateRecord,
    /// Parsed Time
    pub time: Option<TimeRecord>,
    /// Parsed `UTCOffset`.
    pub offset: Option<UtcOffset>,
    /// Parsed `TimeZone` data (UTCOffset | IANA name)
    pub tz: Option<TimeZone<'a>>,
    /// The parsed calendar value.
    pub calendar: Option<&'a str>,
    /// Any additional annotations beyond time zone and calendar
    pub annotations: Vec<Annotation<'a>>,
}
```

Reply:
Yes, as far as `ixdtf` is concerned, returning `offset` and `tz` as separate fields sounds like a great idea.